src/Lecture3.hs

```diff
@@ -150,16 +161,20 @@ data Treasure a
     deriving (Show, Eq)
 
 {- | When you append multiple treasures for fighting multiple
-monsters, you should get a combined treasure and not just the first
-(or last one).
+  monsters, you should get a combined treasure and not just the first
+  (or last one).
 
 🕯 HINT: You may need to add additional constraints to this instance
   declaration.
 -}
-instance Semigroup (Treasure a) where
+instance (Semigroup a) => Semigroup (Treasure a) where
+    (<>) NoTreasure a = a
+    (<>) b NoTreasure = b
+    (<>) (SomeTreasure a) (SomeTreasure b) = SomeTreasure (a <> b)
+    
 
-
-instance Monoid (Treasure a) where
+instance (Monoid a) => Monoid (Treasure a) where
```

Comment:
The `Monoid` constraint is too strict here. You can relax it to `Semigroup` as you don't use `mempty` from `Monoid` here 🙂 It will make this instance usable in more situations.
```suggestion
instance (Semigroup a) => Monoid (Treasure a) where
```

Reply:
ok )